### ISSUE TITLE
kml2gmt failing

### ISSUE THREAD

**Issue by macpurity** [GenericMappingTools/gmt#1012]
**Description of the problem**

Simply want to make a quick conversion from kml to GMT. Using the command pasted below.

Yields only the initial header record, then nothing else (no coordinates). File attached. Is it a file issue or a read-in issue?

**Full script that generated the error**

```
kml2gmt IS2_RGT_0002_cycle4_28-Jun-2019.kml -Fl
```

**Full error message**

```
None - even with -Vd. Just outputs the header, then nothing else.
```

**System information**

* Operating system: OSX 10.13.6
* Version of GMT: Version 6.1.0_908ab21_2019.06.27 [64-bit] [8 cores]

**Comment by joa-quim** [comment]
Please, try again to attach the file. There is none.

**Comment by macpurity** [comment]
Did it work, that time? I always have trouble uploading files!

**Comment by joa-quim** [comment]
Nope. Still no file. 
It's very easy. Just drag-n-drop the file inside the comment box.

**Comment by macpurity** [comment]
That was what I did. It won't "stick" when dropping a file into the comment box. Maybe something in my GitHub user settings prevents me?

**Comment by seisman** [comment]
GitHub doesn't allow uploading a kml file. You need zip it.

![image](https://user-images.githubusercontent.com/3974108/60281838-b1cc7a00-98d3-11e9-836a-362d75335e96.png)

**Comment by macpurity** [comment]
It has been zipped (or gzipped). I think it may be a setting in my browser. Trying again...

Yes. It was a security setting in my browser. Here we go!
[IS2_RGT_0002_cycle4_28-Jun-2019.kml.gz](https://github.com/GenericMappingTools/gmt/files/3335646/IS2_RGT_0002_cycle4_28-Jun-2019.kml.gz)

**Comment by joa-quim** [comment]
Here, it prints the header and then apparently falls into an infinite (or really long) loop. GDAL fails too
```
ogr2ogr -f "ESRI shapefile" lolo.shp IS2_RGT_0002_cycle4_28-Jun-2019.kml
ERROR 1: Failed to create file lolo.shp\RGT 2 DOY-179 28-Jun-2019 04:55:18.shp: No error
ERROR 1: Terminating translation prematurely after failed
translation of layer RGT 2 DOY-179 28-Jun-2019 04:55:18 (use -skipfailures to skip errors)
```

**Comment by PaulWessel** [comment]
It has a single line with all the coordinates, totaling 219463 characters, so I am guessing related to that?

**Comment by PaulWessel** [comment]
The record length defeats our code.  It starts with

`<coordinates>-23.74810710695745,0.007072056629070391,0 -23.75450979455047,... 
`

and finally ends with

` ,0 -47.48800222151229,-0.08420427733283828,0 </coordinates>
`

 So a single line with all the data, but it is way too long since we are reading in a single line with GMT_BUFSIZ.  So the test that looks for </coordinates> does not find it because of truncation, and all hell breaks loose.

Two solutions to this problem:

1. Just use a very large string array for this to anticipate the occasional very long lines.  We do this for other i/o, e.g., allocate GMT_INITIAL_MEM_ROW_ALLOC doubles for data so this is truly the easy solution.
2. Completely rewrite the input reading to parse such silly files.

I vote for the first, i.e set char line[GMT_INITIAL_MEM_ROW_ALLOC] and use GMT_INITIAL_MEM_ROW_ALLOC in those fgets lines.

**Comment by joa-quim** [comment]
Yes. For sure sloppy programmers that think memory is always infinite.

**Comment by PaulWessel** [comment]
GMT_INITIAL_MEM_ROW_ALLOC < oo

**Comment by macpurity** [comment]
Thank you guys, for looking into this. Google Earth is able to ingest this file. Is this going to be a problem for GMT? I can ask the person who created the file to remake it in a different way, if I knew what to ask.

**Comment by joa-quim** [comment]
In this case, to make it work with current GMT, you can ask him to not to write all coordinates is a single line. GMT is reading lines at most with 4096 chars. Note, we can have many lines, but each one cannot be larger than 4 Kb. This will change, but if you can ask to change the way the file is written than current GMT should be able to ingest it.

**Comment by joa-quim** [comment]
And another thing. This KML file has 96 layers, but 95 are empty.

**Comment by seisman** [comment]
Fixed by #1057
